src/snowflake/connector/errorcode.py

```diff
@@ -16,6 +16,7 @@
 ER_UNABLE_TO_OPEN_BROWSER = 250008
 ER_UNABLE_TO_START_WEBSERVER = 250009
 ER_INVALID_CERTIFICATE = 250011  # not used but keep here to reserve errno
+ER_INVALID_BACKOFF_POLICY = 250012
```

Comment:
I've searched for these error codes throughout confluence and on public documentation, and I haven't been able to find anything. I also noticed some errors like:

```
ER_NOT_SUPPORT_DATA_TYPE = 255001
ER_NO_PYARROW = 255002
ER_NO_ARROW_RESULT = 255003
ER_NO_PYARROW_SNOWSQL = 255004
ER_FAILED_TO_READ_ARROW_STREAM = 255005
ER_NO_NUMPY = 255006
```

Which seem Python specific. Also, errors like `ER_NO_ACCOUNT_NAME` will never reach the server or come back from the server since it is just a client side check:

<img width="711" alt="image" src="https://github.com/snowflakedb/snowflake-connector-python/assets/144391642/7fd4fb9c-940c-4a43-9ef7-baac5a5003f1">

So I figured I could add some codes of my own since none of the existing ones really make sense for my purposes.